cmd/cmd.go

```diff
@@ -33,3 +51,221 @@ const msgMissingMetadata = `
     Please see https://exercism.io/cli-v1-to-v2 for instructions on how to fix it.
 
 `
+
+type downloadContext struct {
+	usrCfg  *viper.Viper
+	uuid    string
+	slug    string
+	track   string
+	team    string
+	payload *downloadPayload
+}
+
+func newDownload(ctx *downloadContext) error {
+	client, err := api.NewClient(ctx.usrCfg.GetString("token"), ctx.usrCfg.GetString("apibaseurl"))
+	if err != nil {
+		return err
+	}
+
+	req, err := client.NewRequest("GET", ctx.requestURL(), nil)
+	if err != nil {
+		return err
+	}
+
+	query := req.URL.Query()
+	ctx.buildQuery(query)
+	req.URL.RawQuery = query.Encode()
+
+	res, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if err := json.NewDecoder(res.Body).Decode(&ctx.payload); err != nil {
+		return fmt.Errorf("unable to parse API response - %s", err)
+	}
+
+	if res.StatusCode == http.StatusUnauthorized {
+		siteURL := config.InferSiteURL(ctx.usrCfg.GetString("apibaseurl"))
+		return fmt.Errorf("unauthorized request. Please run the configure command. You can find your API token at %s/my/settings", siteURL)
+	}
+
+	if res.StatusCode != http.StatusOK {
+		switch ctx.payload.Error.Type {
+		case "track_ambiguous":
+			return fmt.Errorf("%s: %s", ctx.payload.Error.Message, strings.Join(ctx.payload.Error.PossibleTrackIDs, ", "))
+		default:
+			return errors.New(ctx.payload.Error.Message)
+		}
+	}
+	return nil
+}
+
+func (d *downloadContext) writeMetadata() error {
```

Comment:
Sometimes we've called this `d` other times we've called it `ctx`. I think it would be worth being consistent.